### ISSUE TITLE
Force the amplitude of a field to be len 3 for vector fields

### ISSUE THREAD

**Issue by gschlafly** [chromatix-team/chromatix#112]
A vectorfield plane wave can be defined as follows:
```python
field = cf.plane_wave(
        (1, 1),
        1.0,
        0.532,
        1.0,
        amplitude=[1, 1, 1],
        scalar=False,
    )
``` 
If the `amplitude=[1, 1]` with `scalar=False` then the field is not compatible with the polarization functions.

**Comment by diptodip** [comment]
Should be possible in a JAX-friendly way with https://chex.readthedocs.io/en/latest/api.html#chex.assert_axis_dimension

**Comment by GJBoth** [comment]
Closed by #138
